backends/printing/printman.h

```diff
@@ -0,0 +1,107 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef BACKENDS_PRINTING_PRINTMAN_H
+#define BACKENDS_PRINTING_PRINTMAN_H
+
+#ifdef USE_PRINTING
+
+#include "common/rect.h"
+#include "common/rational.h"
+#include "common/scummsys.h"
+#include "common/str.h"
+#include "common/file.h"
+#include "graphics/managed_surface.h"
+
+class PrintJob;
+
+class PrintingManager {
+public:
+	virtual ~PrintingManager();
+
+	virtual PrintJob *createJob(const Common::String &jobName) = 0;
```

Comment:
Could you please explain the reasoning behind this PrintJob class?

Reply:
I believe I already did? Let me try again then.

The PrintingManager class is the central starting point for the feature of printing. It is where things such as printer enumeration should go.

The PrintJob class represents an active print job that is being spooled. The user has decided to print and with what settings to do so.

Reply:
But there is nothing here that can't be done in PrintJob.
This would make the implementation way simpler.